api/v1alpha1/mariadb_webhook.go

```diff
@@ -201,3 +202,16 @@ func (r *MariaDB) validatePodDisruptionBudget() error {
 	}
 	return nil
 }
+
+func (r *MariaDB) validateStorage() error {
+	logger.V(1).Info("Validate MariaDB storage", "mariadb", r.Name)
+
+	if !r.IsEphemeralStorageEnabled() && r.Spec.VolumeClaimTemplate == nil {
```

Comment:
You can compare `r.Spec.VolumeClaimTemplate` with an empty struct.

Reply:
Done using the reflect package, I'm not a go expert but comparing nested structs with slices seems tricky.

Reply:
Good shout, I was thinking of:
```go
r.Spec.VolumeClaimTemplate == (corev1.PersistentVolumeClaim{})
```
but you are right, the compiler complains if the struct has slices